store/src/java/com/zimbra/cs/imap/ImapHandler.java

```diff
@@ -877,6 +886,14 @@ boolean executeRequest(ImapRequest req) throws IOException, ImapException {
                     req.skipSpace();  Set<String> patterns = Collections.singleton(req.readFolderPattern());
                     checkEOF(tag, req);
                     return doLIST(tag, base, patterns, (byte) 0, RETURN_XLIST, (byte) 0);
+                } else if (command.equals("X-ZIMBRA-FLUSHCACHE")) {
+                    req.skipSpace();
+                    CacheEntryType[] cacheTypes = req.readCacheEntryTypes();
+                    CacheEntry[] entries = req.readCacheEntries();
+                    return doFLUSHCACHE(tag, cacheTypes, entries);
```

Comment:
Seems like a checkEOF(tag, req) is appropriate here just in case extra text is left in req after the list entries (if any).